### PR TITLE
Update DG use cases and add planned enhancement for duplicate detection.

### PR DESCRIPTION
**Pull request by CJerrong**
- Add use cases for `edit`, `logedit`, `undo` and `redo`
- Refactor use cases in DG to omit mechanical details like the use of index
- Add planned enhancement for better duplicate detection, to warn users when inputting similar values with minor variations like extra white spacings.
- Add in page break formatting for DG
- Fix typos in DG

### CODE REVIEW

**Review by gerteck — APPROVED** [review]
looks good 👍 

But don't merge yet, can add the page breaks before merging

**Review by soons1 — COMMENTED** [review]
LGTM 👍

**Review by sdevih — COMMENTED** [review]
LGTM!

**Pull request merged**
